dist/setup/index.js

```diff
@@ -94663,8 +94663,14 @@ function parseGoVersionFile(versionFilePath) {
     const contents = fs_1.default.readFileSync(versionFilePath).toString();
     if (path.basename(versionFilePath) === 'go.mod' ||
         path.basename(versionFilePath) === 'go.work') {
-        const match = contents.match(/^go (\d+(\.\d+)*)/m);
-        return match ? match[1] : '';
+        // toolchain directive: https://go.dev/ref/mod#go-mod-file-toolchain
+        const matchToolchain = contents.match(/^toolchain go(\d+(\.\d+)*)/m);
```

Comment:
That regex matches
```
toolchain go1       # invalid (major isn't allowed)
toolchain go1.2.3.4 # invalid (not minor or semver)
```
but not
```
toolchain go1.25rc1 # valid (prerelease)
```
https://go.dev/doc/toolchain

Better:
```suggestion
        const matchToolchain = contents.match(^toolchain go(1\.\d+(?:\.\d+|rc\d+)?));
```

Reply:
> That regex matches
> 
> ```
> toolchain go1       # invalid (major isn't allowed)
> toolchain go1.2.3.4 # invalid (not minor or semver)
> ```
> 
> but not
> 
> ```
> toolchain go1.25rc1 # valid (prerelease)
> ```
> 
> https://go.dev/doc/toolchain
> 
> Better:

See b967a467e1c904489be015f76916b6bb83d60ef2, the check on the `go` directive should probably use the same pattern, but I think that can be a separate change